blockchain/event_filter.go

```diff
@@ -149,7 +162,13 @@ func (e *EventFilter) Events(cToken *ContinuationToken, chunkSize uint64) ([]*Fi
 			return nil, nil, err
 		}
 	}
-	return matchedEvents, nil, nil
+
+	// Create a continuation token if there are remaining blocks
+	var rToken *ContinuationToken
+	if remainingScannedBlocks == 0 {
+		rToken = &ContinuationToken{fromBlock: curBlock, processedEvents: processedEvents}
+	}
+	return matchedEvents, rToken, nil
```

Comment:
We can consolidate the early return above in to this as well. They are essentially returning the same thing.
Let's keep this one and remove the previous one.